using-mobile-extensions/adobe-experience-platform-messaging/messaging-api-reference.md

```diff
@@ -0,0 +1,420 @@
+
+#### Examples
+
+```swift
+func application(_: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+    MobileCore.setPushIdentifier(deviceToken)
+}
+```
+
+### Objective-C
+
+#### Syntax
+
+```objective-c
+public static func setPushIdentifier(_ deviceToken: Data?)
+```
+
+#### Examples
+
+```objective-c
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken { 
+    [AEPMobileCore setPushIdentifier:deviceToken];
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## addPushTrackingDetails<a id="addpushtrackingdetails"></a>
+
+This API is used to updated the push notification pending intent with necessary customer journey information.
+{% hint style="warning" %}
+Call to this API is necessary to ensure that all the important information (messageId, Customer journey information) are added to the pending intent so that they can be used while tracking the push notification interactions.
+{% endhint %}
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+#### Syntax
+
+```java
+public static boolean addPushTrackingDetails(final Intent intent, final String messageId, final Map<String, String> data)
+```
+
+* *intent* - `Intent` which is added to the pending intent so that it can be used when user interacts with the notification.
+* *messageId* - `String` message ID of the push notification
+* *data* - `Map` which represents the data part of the remoteMessage.
+* Returns `boolean` indicating whether the intent was update with necessary information (messageId and customer journey data).
+
+
+#### Examples
+
+```java
+boolean update = addPushTrackingDetails(intent, messageId, data)
+```
+
+{% endtab %}
+{% endtabs %}
+
+## Public Classes
+### AEPMessagingFCMPushPayload
+
+This is a helper class for extracting the data payload attributes from `RemoteMessage` which are used while creating the push notification.
+Create the instance of `AEPMessagingFCMPushPayload` in the `onMessageReceived` method using the below constructors.
+
+### Constructor
+
+{% tabs %}
+{% tab title="Android" %}
+### Java
+
+#### Syntax
+
+```java
+public AEPMessagingFCMPushPayload(RemoteMessage message)
+```
+
+* *message* - `RemoteMessage` message containing the payload data with the necessary attributes for creating push notification
+
+```java
+public AEPMessagingFCMPushPayload(Map<String, String> data)
+```
+
+* *data* - `Map<String, String>` data payload containing the necessary attributes for creating push notification
+
+#### Examples
+
+```java
+// Using the remote message
+@Override
+public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
+    AEPMessagingFCMPushPayload payload = new AEPMessagingFCMPushPayload(remoteMessage);
+}
+
+// Using the data map
+@Override
+public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
+    AEPMessagingFCMPushPayload payload = new AEPMessagingFCMPushPayload(remoteMessage.getData());
+}
+```
```

Comment:
it's not clear what the point of the alternative constructor is

Reply:
If someone does not want use to see all the fields from remote message.